packages/vite/src/node/server/pluginContainer.ts

```diff
@@ -285,10 +283,22 @@ export async function createPluginContainer(
           : // some rollup plugins, e.g. json, sets position instead of pos
             (err as any).position
       if (pos != null) {
-        err.loc = err.loc || {
-          file: err.id,
-          ...numberToPos(ctx._activeCode, pos)
+        let errLocation;
+        try {
+          errLocation = numberToPos(ctx._activeCode, pos);
         }
+        catch (err2) {
+          logger.error(
+            chalk.red(`Error in error handler:\n${err2.stack || err2.message}`),
+            { error: err2 }
+          )
+          // TODO verify that err is visible
+          throw err;
+        }
```

Comment:
> TODO verify that err is visible

Reply:
how to add a unit test for this?
vite.createServer -> createPluginContainer -> formatError

broken code is passed to the plugin -> plugin throws error1
plugin returns a broken sourcemap -> numberToPos throws error2 "offset is longer than source length"

Reply:
@patak-js Do we even test logged errors?
I'm not sure if we need to test this :thinking: